spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -283,23 +291,31 @@ public void testUseSnapshotIdForTagReferenceAsOf() {
   }
 
   @Test
-  public void testBranchReferenceAsOf() {
+  public void testBranchReference() {
     Table table = validationCatalog.loadTable(tableIdent);
     long snapshotId = table.currentSnapshot().snapshotId();
     table.manageSnapshots().createBranch("test_branch", snapshotId).commit();
-
-    // create a second snapshot, read the table at the snapshot
     List<Object[]> expected = sql("SELECT * FROM %s", tableName);
+
+    // create a second snapshot, read the table at the branch
     sql("INSERT INTO %s VALUES (4, 'd', 4.0), (5, 'e', 5.0)", tableName);
     List<Object[]> actual1 = sql("SELECT * FROM %s VERSION AS OF 'test_branch'", tableName);
     assertEquals("Snapshot at specific branch reference name", expected, actual1);
 
-    // read the table at the snapshot
+    // read the table at the branch
     // HIVE time travel syntax
     List<Object[]> actual2 =
         sql("SELECT * FROM %s FOR SYSTEM_VERSION AS OF 'test_branch'", tableName);
     assertEquals("Snapshot at specific branch reference name", expected, actual2);
 
+    // Spark session catalog does not support extended table names
+    if (!"spark_catalog".equals(catalogName)) {
+      // read the table at the branch
```

Comment:
```suggestion
      // read the table using the "branch_" prefix in the table name
```

Reply:
can you also please apply the same suggestions to the 3.3/3.4 files?

Reply:
Done.